ludwig/trainers/trainer.py

```diff
@@ -1035,7 +1043,7 @@ def train(
             if test_summary_writer is not None:
                 test_summary_writer.close()
 
-            if not self.skip_save_model and self.skip_all_evaluation:
+            if not self.skip_save_model and self.skip_all_evaluation and not has_nan_or_inf_tensors:
```

Comment:
I think yes - there's no point of continuing training for now since subsequent epochs will have NaN loss. In the future, we can think of other mechanisms, such as lowering learning rate, decreasing the batch size / gradient accumulation steps, etc from the last valid checkpoint!